packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx

```diff
@@ -99,34 +101,43 @@ export function LocationsView({
       hasError,
     });
 
+  const contextValue: ControlsContext = {
+    data: {
+      pagination: {
+        currentPage,
+        disableNext,
+        disablePrevious,
+        handlePaginateNext: () =>
+          handlePaginateNext({ resultCount, hasNextToken }),
+        handlePaginatePrevious: () =>
+          handlePaginatePrevious({ resultCount, hasNextToken }),
+      },
+    },
+    actionsConfig: { type: 'LIST_LOCATIONS', isCancelable: false },
+  };
+
   return (
-    <div
-      className={resolveClassName(CLASS_BASE, className)}
-      data-testid="LOCATIONS_VIEW"
-    >
-      <Title>Home</Title>
-      <RefreshControl
-        disableRefresh={disableRefresh}
-        handleRefresh={() => {
-          handleReset();
-          handleList({
-            options: { ...DEFAULT_LIST_OPTIONS, refresh: true },
-          });
-        }}
-      />
-      <Paginate
-        currentPage={currentPage}
-        disableNext={disableNext}
-        disablePrevious={disablePrevious}
-        handleNext={() => {
-          handlePaginateNext({ resultCount, hasNextToken });
-        }}
-        handlePrevious={handlePaginatePrevious}
-      />
-      <LocationsMessage />
-      <Loading />
-      <DataTableControl range={range} />
-      <LocationsEmptyMessage />
-    </div>
```

Comment:
The only changes in lines 103 - 130 are wrapping everything in the `<ControlsContextProvider>` and replacing `<Paginate>` with `<PaginationControl>`